lib/targets.py

```diff
@@ -0,0 +1,41 @@
+import os
+import shutil
+
+from luigi import Target, LocalTarget
+
+from tasks.util import classpath
+
+
+class DirectoryTarget(Target):
+    '''
+    Similar to :class:`LocalTarget <luigi.LocalTarget>` but automatically creates and destroys temp directories
+    '''
+    def __init__(self, task):
+        self.path = os.path.join('tmp', classpath(task), task.task_id)
```

Comment:
As this class is `DirectoryTarget` and not `DODirectoryTarget` I would pass the `path` as the parameter instead of the `task`

Reply:
This is a pattern I have seen in a lot of tasks, and I prefer discharging the task of this, since the classpath thing will be needed in many places.

I agree that passing the task here is not the cleanest though, so open to suggestions. I thought about doing a small mixin class for tasks. Something like:
```python
class FileSystemTask:
  def output_path(self):
    return os.path.join('tmp', classpath(self), self.task_id)

class DownloadUK(DownloadUnzipTask, FileSystemTask):
  def output():
    return DirectoryTarget(path=self.output_path())
```

But it seems more complicated and I don't know if it is very useful. That's why I went with the current approach, and we can refine it as we add more tasks using it.

Reply:
It's not so horrible, I was just thinking out loud